### PR TITLE
Round end zombie percentage changes

### LINKED ISSUE #15574: End of round zombie infection percentage is innacurate
## Description
Lizard round #23820 showed 33% of crew infected when there were under 8 surviving crewmembers on a likely nearly full server.

**Reproduction**
1. I don't know

### PR DESCRIPTION
**Pull request by Slava0135**
## About the PR

closes #15574

I think previous formula is confusing for most people as it only counts alive zombies and humans. With these changes it counts all human bodies (however you can gib them, so maybe its not the best solution).

**Media**

Peek 2023-04-21 10-23

**Changelog**

:cl:
- tweak: round end zombie percentage is now calculated as = (infected humanoid bodies) / (total humanoid bodies)
- tweak: round end zombie survivor list now shows more survivors

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
Obligatory: dont merge until gamerule entities merged.

**Comment by ElectroJr** [comment]
Just counting all humanoid bodies seems like a bad idea because it'll just get skewed by misc bodies from other sources which have nothing to do with the initial number of player-controlled humanoids on the station. The game rule probably just has to be changed so that it properly keeps track of player-controlled humans (after the current freeze is over)

**Comment by Slava0135** [comment]
> Just counting all humanoid bodies seems like a bad idea because it'll just get skewed by misc bodies from other sources which have nothing to do with the initial number of player-controlled humanoids on the station. The game rule probably just has to be changed so that it properly keeps track of player-controlled humans (after the current freeze is over)

- arriving - no problem here
- cloning - usually you grind old body 
- salvage - bodies are quite rare, and likely people won't do salvage in zombie outbreak

Ideally, you would track of player-controlled humans somehow but I think this implementation is decent enough (though it may deviate due to body gibbing and etc.). At least it concludes round end better.

File: Content.Server/GameTicking/Rules/ZombieRuleSystem.cs
Comment (AJCM-git) [inline Content.Server/GameTicking/Rules/ZombieRuleSystem.cs:200]: why query HumanoidAppearanceComponent, its unnecesary and it looks like you only need ZombieComponent here, also take the queries as parameters since you repeat them for GetHealthyHumans so you can feed the queries to both methods if you make em on OnRoundEndText handler

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Server/GameTicking/Rules/ZombieRuleSystem.cs
Comment (Slava0135) [reply Content.Server/GameTicking/Rules/ZombieRuleSystem.cs:200]: HumanoidAppearance is here because we only count humanoids

Comment (Slava0135) [reply Content.Server/GameTicking/Rules/ZombieRuleSystem.cs:200]: also, its only run once when round ends, it shouldn't be a big performance hit for not reusing queries (they are different even)

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**
